src/misc/cross.cpp

```diff
@@ -715,6 +715,88 @@ std::string cfstr_to_string(CFStringRef source)
 	return lang;
 }
 
+#if defined (__linux__)
+
+#include <fcntl.h>
+
+int get_num_physical_cpus()
+{
+	constexpr int MaxCpus = 128;
+	bool core_ids[MaxCpus] = {};
+	int num_cores = 0;
+	for (int i = 0; i < MaxCpus; ++i) {
+		std::string path = "/sys/devices/system/cpu/cpu" + std::to_string(i) + "/topology/core_id";
+		int fd = open(path.c_str(), O_RDONLY);
+		if (fd != -1) {
+			char core_id_str[6] = {};
+			if (read(fd, core_id_str, 5) > 0) {
+				int core_id = atoi(core_id_str);
+				if (core_id >= 0 && core_id < MaxCpus && !core_ids[core_id]) {
+					core_ids[core_id] = true;
+					++num_cores;
+				}
+			}
+			close(fd);
+		}
+	}
+	return std::max(num_cores, 1);
```

Comment:
minor: some newlines & clang format would improve the readability slightly

Reply:
Fixed